server/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java

```diff
@@ -2445,4 +2449,65 @@ year TIMESTAMP GENERATED ALWAYS AS date_trunc('year',ts)
             .as("Column values must match: " + Arrays.toString(response.rows()[0]))
             .isEqualTo(response.rows()[0][1]);
     }
+
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    @UseRandomizedOptimizerRules(value = 0)
+    public void to_be_removed() {
+        execute("""
+            CREATE TABLE IF NOT EXISTS "doc"."weather_data_partitioned"(
+               "timestamp" TIMESTAMP WITHOUT TIME ZONE,
+               "ts_month"   GENERATED ALWAYS AS date_trunc('month', "timestamp")
+            ) CLUSTERED INTO 1 SHARDS
+            PARTITIONED BY ("ts_month") with (number_of_replicas = 0);
+            """
+        );
+
+        // No replicas, each partition has 1 shard, total number of shards is 3.
+        execute("""
+            INSERT INTO weather_data_partitioned (timestamp)
+            VALUES ('2023-01-01 00:00:40'), ('2023-02-01 00:00:40'), ('2023-03-01 00:00:40')
+            """
+        );
+
+        // Hits 2 partitions (2 shards) out of total 3.
+        execute("""
+            EXPLAIN ANALYZE SELECT ts_month
+            FROM "doc"."weather_data_partitioned"
+            WHERE "timestamp" >= '2023-01-01' AND "timestamp" < '2023-03-01'
+           """
+        );
+
+        int shardsHit = shardsHit(response);
+        // Should be 2? WhereClauseOptimizerTest shows that filter is expanded.
+        // UseRandomizedOptimizerRules is enabled so shouldn't affect.
+        assertThat(shardsHit).isEqualTo(3);
+
+
+        // No optimization is needed, hits 2 shards out of total 3.
+        execute("""
+            EXPLAIN ANALYZE
+            SELECT ts_month
+            FROM "doc"."weather_data_partitioned"
+            WHERE "ts_month" >= '2023-01-01' AND "ts_month" < '2023-03-01'
+            """
+        );
+        shardsHit = shardsHit(response);
+        assertThat(shardsHit).isEqualTo(2);
+    }
+
+    private static int shardsHit(SQLResponse response) {
+        Map<String, Object> analysis = (Map<String, Object>) response.rows()[0][0];
+        Map<String, Object> executeAnalysis = (Map<String, Object>) analysis.get("Execute");
+
+        int shardsHit = 0;
+        for(Object val: executeAnalysis.values()) {
+            if (val instanceof Map m && m.get("QueryBreakdown") != null) {
+                shardsHit += ((List) m.get("QueryBreakdown")).size();
+            }
+        }
+        return shardsHit;
```

Comment:
The whole integration test is subject of removal. This is ugly but doesn't matter as it's throwaway code.